tensorflow/core/platform/default/logging.cc

```diff
@@ -392,5 +345,182 @@ bool LogEveryNSecState::ShouldLog(double seconds) {
   return true;
 }
 
+std::mutex TFLogSinks::instanceMutex_;
+std::atomic<TFLogSinks*> TFLogSinks::instance_ = ATOMIC_VAR_INIT(nullptr);
+
+TFLogSinks& TFLogSinks::Instance() {
+  auto instance = instance_.load(std::memory_order_acquire);
+  if(!instance) {
+    std::lock_guard<std::mutex> lock(instanceMutex_);
+    instance = instance_.load(std::memory_order_relaxed);
+    if(!instance) {
+      instance = new TFLogSinks;
+      instance_.store(instance, std::memory_order_release);
+    }
+  }
+
+  return *instance;
+}
+
+void TFLogSinks::Add(TFLogSink* sink) {
+  if(sink == nullptr) {
```

Comment:
If user passes a nullptr, it's likely a bug and it'll be good to be discovered early.  So let's remove the check and let it crash for nullptr.  Same for `TFLogSinks::Remove`.

Reply:
Will, do. That's really not how we do things, but then again we use exceptions, while you don't.